diff --git a/arch/arm64/boot/dts/rockchip/lec-px30.dtsi b/arch/arm64/boot/dts/rockchip/lec-px30.dtsi
index 708bc9cd419d..7fc38283b290 100644
--- a/arch/arm64/boot/dts/rockchip/lec-px30.dtsi
+++ b/arch/arm64/boot/dts/rockchip/lec-px30.dtsi
@@ -2120,7 +2120,7 @@
 			};
 		};
 
-		uart3-m0 {
+		/*uart3-m0 {
 			uart3m0_xfer: uart3m0-xfer {
 				rockchip,pins =
 					<0 RK_PC0 RK_FUNC_2 &pcfg_pull_up>,
@@ -2141,7 +2141,7 @@
 				rockchip,pins =
 					<0 RK_PC3 RK_FUNC_GPIO &pcfg_pull_none>;
 			};
-		};
+		};*/
 
 		uart3-m1 {
 			uart3m1_xfer: uart3m1-xfer {
